sympy/core/tests/test_arit.py

```diff
@@ -187,6 +187,13 @@ def test_pow3():
     assert sqrt(2)**3 == sqrt(8)
 
 
+def test_mod_pow():
+    assert pow(S(4), 13, 497) == pow(4, 13, 497)
+    assert pow(S(4), -3, 497)*4**3%497 == 1
+    assert pow(x, y, z) == x**y%z
```

Comment:
Can you test all combinations of SymPy/Python input types? I don't remember exactly how Python does dispatch for three-argument `__pow__`, but we should make sure all the combinations are tested.

Reply:
You might also add a test for something like `pow(S(2), S(10000000000), S(3))` (you can add more zeros if necessary), to make sure that it is using the efficient algorithm.

Reply:
@asmeurer I'm not sure how and whether to test that it is using efficient algorithm, since it depends on Python implementation.

Reply:
`pow(S(2), S(10000000000), S(3))` will do the trick. If you try to evaluate `2**10000000000 % 3` naively it won't work. Using the Python implementation is fine. This just tests that it does indeed do it (or something equivalent).

Reply:
@asmeurer Added tests for various types 😃